app/services/proxy_status/logs.rb

```diff
@@ -0,0 +1,18 @@
+module ProxyStatus
+  class Logs < Base
+    def api
+      ProxyAPI::Logs.new(:url => proxy.url)
```

Comment:
That's what `api_class` is for in base - I had the same problem with puppet ca.
If you take a look at the api function in base, you will notice that it actually does two things except the api call - one is reusing the api instance object if it already exists instead of creating a new one for every api call, and the second is catching any NameErrors raised by the initialization of the api object.